src/mkdir.js

```diff
@@ -51,29 +51,13 @@ function mkdirSyncRecursive(dir) {
 function _mkdir(options, dirs) {
   if (!dirs) common.error('no paths given');
 
-  if (typeof dirs === 'string') {
```

Comment:
I think the type is always string. We introduced some code to flatten out arrays. Even in the case of ShellStrings, I think we also convert those into bare strings.